### PR TITLE
Allow quoted cookie values, as per RFC 6265

### PR DESCRIPTION
**Pull request by wolfsage**
The spec says:

  cookie-value      = *cookie-octet / ( DQUOTE *cookie-octet DQUOTE )

### CODE REVIEW

File: lib/Plack/Request.pm [hunk 1]
```diff
@@ -68,7 +68,15 @@ sub cookies {
         # trim leading trailing whitespace
         $pair =~ s/^\s+//; $pair =~ s/\s+$//;
 
-        my ($key, $value) = map URI::Escape::uri_unescape($_), split( "=", $pair, 2 );
+        my ($key, $value) = split( "=", $pair, 2 );
+
+        $key = URI::Escape::uri_unescape($key);
+
+        # Values can be quoted
+        $value =~ s/\A"//;
```
Comment (miyagawa) [inline lib/Plack/Request.pm]: i don't mean to be pedantic but should we only unquote if the quotes are balanced?

Comment (wolfsage) [inline lib/Plack/Request.pm]: I'm not sure there's a "right" answer here, but what's the path of least surprise?

Comment (miyagawa) [inline lib/Plack/Request.pm]: like, what could be a problem if we do: `$value =~ s/^"(.*)"$/$1/;` ?

Comment (wolfsage) [inline lib/Plack/Request.pm]: I can't think of any, so that works for me I guess.

Comment (wolfsage) [inline lib/Plack/Request.pm]: Updated with requested behaviour

**Comment by miyagawa** [comment]
Is it possible to add a test that sets these value using Plack::Response (via Cookie::Baker) and reads them back, since that would guarantee that we're doing symmetric read/write for the data?

**Comment by miyagawa** [comment]
Actually, would it make more sense if we use Cookie::Baker's `crush_cookie` function? cc @kazeburo

**Comment by kazeburo** [comment]
@miyagawa @wolfsage 
I sent PR to Cookie::Baker https://github.com/kazeburo/Cookie-Baker/pull/8
Please review it.

**Comment by miyagawa** [comment]
@kazeburo are you going to make a PR against Plack to make use of that in Plack::Request?

**Comment by kazeburo** [comment]
@miyagawa Yes. I'll make a PR.

**Comment by miyagawa** [comment]
closing in favor of #567

**Pull request closed without merging**